### ISSUE TITLE
Difference between documentation and code in Sobol indices 

### ISSUE THREAD

**Issue by MerlinKeller** [openturns/openturns#2260]
From a colleague :

-----------------------------------------------------------
En suivant le lien du code OpenTURNS, j’ai remarqué qu’il y a deux différences entre la documentation et le code :

    L’estimateur Saltelli de l’indice d’ordre total STi :
        Dans la doc d’OpenTURNS [1] : 

![image](https://user-images.githubusercontent.com/43726507/228591428-79764e40-20ee-47dc-9730-e71b5558b062.png)

        Dans SaltelliSensitivityAlgorithm.cxx [2], ligne 97 : 

![image](https://user-images.githubusercontent.com/43726507/228591667-db05a4fd-a336-4839-a515-e2f237cfab3c.png)
 
à Et donc on utilise « mean(g(A))*mean(g(A)) » au lieu de « mean(g(A))*mean(g(B)) » dans « V_{-i} ».

    L’estimateur Jansen de l’indice de premier ordre Si :
        Dans la doc d’OpenTURNS [3] : 

![image](https://user-images.githubusercontent.com/43726507/228591903-8810bc52-455a-49ef-b042-a57a62c8d6dc.png)
 
        Dans JansenSensitivityAlgorithm.cxx [4], ligne 110 : 
 
![image](https://user-images.githubusercontent.com/43726507/228594252-8b13a3ba-3fbb-43ee-9c4f-6939f3bfa1b4.png)

Et donc on utilise « referenceVariance_ » qui est égale à « Var(g(A)) » au lieu de « 1/(N-1) * sum(g(A)²) » dans  « V_{-i} ».

**Comment by mbaudin47** [comment]
Thank you for the accurate bug report. I think, however, that there are confusions, inversions and lack of focus in the bug report, so I try to rephrase it to make the discussion as clear as possible.

In `SaltelliSensitivityAlgorithm`, the doc for V_{-i} is:

![image](https://user-images.githubusercontent.com/31351465/229188152-1a206c35-6d0e-4005-ab90-fc6934545875.png)

but the code is:

https://github.com/openturns/openturns/blob/b6052f35df83a8a8c60ac16794ed55d34a2cdf7f/lib/src/Uncertainty/Algorithm/Sensitivity/SaltelliSensitivityAlgorithm.cxx#L97

Conclusion: The doc uses « mean(g(A)) * mean(g(B)) » and the code uses « mean(g(A)) * mean(g(A)) ».

In `JansenSensitivityAlgorithm`, the doc for V{i} is:

![image](https://user-images.githubusercontent.com/31351465/229189141-2df1fa03-e5c5-4c48-9095-0f58ac204a02.png)

but the code is:

https://github.com/openturns/openturns/blob/b6052f35df83a8a8c60ac16794ed55d34a2cdf7f/lib/src/Uncertainty/Algorithm/Sensitivity/JansenSensitivityAlgorithm.cxx#L110

Conclusion: The doc is « 1 / (N - 1) * sum(g(A)²) » but the code uses `referenceVariance_`.

According to:

https://github.com/openturns/openturns/blob/b6052f35df83a8a8c60ac16794ed55d34a2cdf7f/lib/src/Uncertainty/Algorithm/Sensitivity/SobolIndicesAlgorithmImplementation.cxx#L888

the `referenceVariance_` contains the sample variance of the aggregated samples of A, B, and the collection of E DoEs:

https://github.com/openturns/openturns/blob/b6052f35df83a8a8c60ac16794ed55d34a2cdf7f/lib/src/Uncertainty/Algorithm/Sensitivity/SobolIndicesAlgorithmImplementation.cxx#L872

I think that this is a statistical bug, because the sample variance should be estimated only on independent observations of the output random vector: this hypothesis is satisfied only for the A and B samples, not including the E DoEs.

Hence, the line:

https://github.com/openturns/openturns/blob/b6052f35df83a8a8c60ac16794ed55d34a2cdf7f/lib/src/Uncertainty/Algorithm/Sensitivity/SobolIndicesAlgorithmImplementation.cxx#L884

is also a statistical bug to me, because the sample mean should be estimated with a i.i.d. sample: this hypothesis is satisfied only for the A and B samples, not including the E DoEs.

In your comment @MerlinKeller, you suggest to center w.r.t. mean([g(A) ; g(B) ; g(E)]), which does not seem right to me.

Now notice that all the suggestions that you and I did are second order corrections. 

- For example, when you suggest to use « mean(g(A)) * mean(g(B)) » instead of « mean(g(A)) * mean(g(A)) » in Saltelli's estimator for V_{-i}, please consider that the output is centered, so that both terms are close to zero. If the total Sobol' index is not close to zero, the covariance should dominate. 
- When you suggest to use « 1 / (N - 1) * sum(g(A)²) » instead of `referenceVariance_` in Jansen's estimator for V{i}, please consider that the output is centered, so that the difference between the two terms is only from the estimator of the mean.

As the bottom line, I would say that we should correct the estimator to exactly match Saltelli and Jansen's estimators in order to match the paper (and that is important), but that will (or should ?) not make any practical difference. In the end, we will not match the paper, because these 2 papers did not mention the centering issue so that the implementation should not be able to exactly match the paper. In fact, Martinez's estimator, which does not require to be centered, is still centered because the centering occurs before all estimators take place. Properly centered (with the statistical bug fix that I mentioned), I do not think that this is detrimental.

**Comment by jschueller** [comment]
summary: 
- the doc must reflect the code
- it's ok to differ from the article if we describe how it differs
